src/Raven.Server/Utils/RawDatabaseRecordExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.ServerWide;
+using Raven.Server.ServerWide;
+using Sparrow.Json;
+
+namespace Raven.Server.Utils;
+
+public static class RawDatabaseRecordExtensions
+{
+    public static Dictionary<string, IndexDefinition> MapReduceIndexes(this RawDatabaseRecord record)
+    {
+        var mapReduceIndexes = new Dictionary<string, IndexDefinition>(StringComparer.OrdinalIgnoreCase);
+
+        if (record.Raw.TryGet(nameof(DatabaseRecord.Indexes), out BlittableJsonReaderObject obj) == false || obj == null)
+            return mapReduceIndexes;
+
+        var propertyDetails = new BlittableJsonReaderObject.PropertyDetails();
+        for (var i = 0; i < obj.Count; i++)
+        {
+            obj.GetPropertyByIndex(i, ref propertyDetails);
+
+            if (propertyDetails.Value == null)
+                continue;
+
+            if (propertyDetails.Value is BlittableJsonReaderObject bjro == false)
+                continue;
+
+            if (bjro.TryGet(nameof(IndexDefinition.Type), out IndexType indexType) == false ||
+                (indexType != IndexType.MapReduce && indexType != IndexType.JavaScriptMapReduce))
```

Comment:
shouldn't this come from extension methods?

Reply:
done